modules/sdk-test/src/lib.rs

```diff
@@ -542,6 +542,20 @@ define_tables! {
         update_by update_pk_connection_id = update_by_a(a),
         delete_by delete_pk_connection_id = delete_by_a(a: ConnectionId),
     } #[primary_key] a ConnectionId, data i32;
+
+    PkSimpleEnum {
+        insert_or_panic insert_pk_simple_enum,
+    } #[primary_key] #[index(direct)] a SimpleEnum, data i32;
+}
+
+#[spacetimedb::reducer]
+fn update_pk_simple_enum(ctx: &ReducerContext, a: SimpleEnum, data: i32) -> anyhow::Result<()> {
+    let Some(mut o) = ctx.db.pk_simple_enum().a().find(&a) else {
+        return Err(anyhow!("row not found"));
+    };
+    o.data = data;
+    ctx.db.pk_simple_enum().a().update(o);
+    Ok(())
```

Comment:
I think this method should be expressible through the macro as `update_by update_pk_simple_enum = update_by_a(a),`. We should also probably include the `delete_by` line, so that all of the `Pk*` tables are defined the same way.